charts/stable/common/values.yaml

```diff
@@ -153,39 +153,37 @@ probes:
       failureThreshold: 30
 
 service:
-  enabled: true
-  type: ClusterIP
-  ## Specify the default port information
-  ## It is adviced not to mix different port protocols on the same service
-  port:
+  main:
+    enabled: true
+    type: ClusterIP
+    ## Specify the default port information
+    ## It is adviced not to mix different port protocols on the same service
     port:
-    ## name defaults to http
-    name:
-    ## Accepts: HTTP, HTTPS, TCP and UDP
-    ## HTTPS and HTTPS spawn a TCP service and get used for internal URL and name generation
-    protocol: HTTP
-    ## Specify a service targetPort if you wish to differ the service port from the application port.
-    ## If targetPort is specified, this port number is used in the container definition instead of
-    ## service.port.port. Therefore named ports are not supported for this field.
-    targetPort:
-    ## Specify the nodePort value for the LoadBalancer and NodePort service types.
-    ## ref: https://kubernetes.io/docs/concepts/services-networking/service/#type-nodeport
-    ##
-    # nodePort:
-  additionalPorts: []
+      port:
+      ## name defaults to http
+      name:
+      ## Accepts: HTTP, HTTPS, TCP and UDP
+      ## HTTPS and HTTPS spawn a TCP service and get used for internal URL and name generation
+      protocol: HTTP
+      ## Specify a service targetPort if you wish to differ the service port from the application port.
+      ## If targetPort is specified, this port number is used in the container definition instead of
+      ## service.port.port. Therefore named ports are not supported for this field.
+      targetPort:
+      ## Specify the nodePort value for the LoadBalancer and NodePort service types.
+      ## ref: https://kubernetes.io/docs/concepts/services-networking/service/#type-nodeport
+      ##
+      # nodePort:
+    additionalPorts: []
```

Comment:
Although this moves the needle slightly to the left, it still keeps the problem square in the center for any additional ports and continues the port.port plague. Instead of this structure, what about a structure that is declarative by nature? e.g.
```yaml
services:
  default:
    enabled: true
    type: ClusterIP
    ports:
      http:
        protocol: http
        port: 80
        # targetPort: 
        # nodePort:
      https:
        protocol: https
        port: 443
        # targetPort: 
        # nodePort:
  arbitraryservice2:
    enabled: true
    ...
```

this would allow you to index and ignore some defined name and use exactly what they want, while removing a name key and only specifying a port: as the actual port and not port.port

Reply:
This just refactors services NOT ports.
Ports is on my todo list still but has nothing to do with the services rework.

Reply:
*note:
I'll put some more priority on the port dict things on my todo list though :)